File: apis/src/main/java/com/beat/apis/performance/application/dto/modify/cast/CastModifyResponse.java
package com.beat.apis.performance.application.dto.modify.cast;

import com.beat.global.support.jackson.CdnImageUrl;

public record CastModifyResponse(Long castId,
								 String castName,
								 String castRole,
								 @CdnImageUrl String castPhoto) {

	public static CastModifyResponse of(Long castId, String castName, String castRole, String castPhoto) {
		return new CastModifyResponse(castId, castName, castRole, castPhoto);
	}
}

File: apis/src/main/kotlin/com/beat/apis/performance/api/response/CastModifyResponse.kt
package com.beat.apis.performance.api.response

import com.beat.apis.performance.application.result.CastResult
import com.beat.global.support.jackson.CdnImageUrl

@ConsistentCopyVisibility
data class CastModifyResponse private constructor(
    val castId: Long?,
    val castName: String?,
    val castRole: String?,
    @field:CdnImageUrl val castPhoto: String?,
) {
    companion object {
        fun from(result: CastResult): CastModifyResponse = CastModifyResponse(result.id, result.name, result.role, result.photo)
    }
}
